Train.py

```diff
@@ -76,14 +76,12 @@
                 print('ctr=' + str(data_index.ctr))
                 print('epoch progress=' + str(100 * data_index.ctr /
                                               len(data_index.all_steer)) + '%')
-
                 if args.display:
                     batch.display()
-
                     plt.figure('loss')
                     plt.clf()  # clears figure
                     loss_record['train'].plot('b')  # plot with blue color
                     loss_record['val'].plot('r')  # plot with red color
                     print_timer.reset()
 
-            batch = Batch.Batch(net)  # Reinitialiize batch
+            batch = Batch.Batch(net)  # Reinitialiize batch. ***Why is this necessary?***
```

Comment:
I took a look, essentially this serves to reinitialize the Tensors for camera_data, metadata, and labels. This is because in the forward pass function, we don't initialize these variables, and just start concatenating too them.

I think a better solution is to take this out of the main training loop, create a clear function in the Batch class and reference it at the beginning of the forward pass function. I will amend this pull request with the required changes, before merging into the main repository.